Small Refactor For the REST API Calls

lilongwe opened the pull request:
I have made a small change to avoid user error to

- use the links from the responses in the REST API calls to navigate 
- this is instead of crafting the urls manually
- this will work for all regions and all tariffs
- fixes small bug in tariff detection in https://github.com/eelmafia/octopus-minmax/pull/39
- tested locally and works well

![Screenshot 2025-03-02 at 11 49 36](https://github.com/user-attachments/assets/685b8083-6bb1-4660-ace3-c1ebb2ae15c7)

eelmafia commented:
Thank you for the PR. If you have the time please rebase and fix the conflicts and I'll get it merged

lilongwe commented on tariff.py:
```diff
@@ -27,8 +29,8 @@ def __str__(self):
+    Tariff("go", "Octopus Go", "Octopus Go", r"-go-", "go", True), # Octopus Go
+    Tariff("agile", "Agile Octopus", "Agile Octopus", r"-agile-", "agile", True), # Octopus Agile
+    Tariff("cosy", "Cosy Octopus", "Cosy Octopus", r"-cosy-", r"cosy-octopus", True), # Octopus Cosy
+    Tariff("flexible", "Flexible Octopus", "Flexible Octopus", r"(?<!go-)var", "", False) # Flexible Octopus
```
Octopus Go also has VAR in the code so this get's confused with Flexible Octopus so better to filter out GO from the search and use regex. This won't scale well as other products also have VAR in the code e.g. Intelligent Octopus Go but it fixes a present bug.

lilongwe commented on main.py:
```diff
@@ -28,13 +28,13 @@ def send_notification(message, title="Octobot"):
     print(message)
 
-    apobj = Apprise()
+    apprise = Apprise()
```
`apobj` is very abstract

lilongwe commented:
> Thank you for the PR. If you have the time please rebase and fix the conflicts and I'll get it merged

Hey @eelmafia , I have updated the PR and added comments. Please merge when you are ready. Thanks

joeShuff commented:
Looks good, gave it a run locally and seems to be fetching data correctly.

eelmafia commented:
Good job catching the bug that caused `Flexible Octopus` to appear as the current tariff! I saw it today and was about to investigate and then I noticed you've taken care of it. 😃
